src/main/java/uk/ac/ebi/embl/api/validation/check/entry/SubmitterAccessionCheck.java

```diff
@@ -52,4 +62,16 @@ && showError(assemblyType)) {
   private boolean showError(AssemblyType assemblyType) {
     return !isIgnoreError() && !EntryUtils.excludeDistribution(assemblyType.getValue());
   }
+
+  /**
+   * Truncates the given string to 30 characters and adds a unique string of 5 characters
+   *
+   * @param submitterAccession
+   * @return
+   */
+  public String truncateAndAddUniqueString(String submitterAccession) {
+    return submitterAccession.substring(0, TRUNCATED_ACCESSION_LENGTH)
+        + "-"
+        + RandomStringUtils.randomAlphanumeric(5).toUpperCase();
```

Comment:
- RandomStringUtils.randomAlphanumeric(5) is not guaranteed to be unique.
- UUIDs are practically unique but very long.
- Why not use a int ordinal as I suggested earlier:
```
private static final int TRUNCATED_ACCESSION_LENGTH = 39;
private static AtomicInteger uniqueAccessionIndex = new AtomicInteger();

public String truncateAndAddUniqueString(String submitterAccession) {
    return submitterAccession.substring(0, TRUNCATED_ACCESSION_LENGTH)
        + "-"
        + uniqueAccessionIndex.incrementAndGet();
}
```

Reply:
Yes, I checked UUID was too long.
We get only one Entry in class and looping of entries are done long behind in the call stack, we might need to pass the index order in case of ordering it as what you mentioned.

One solution is to use this